lib/ui/semantics.dart

```diff
@@ -927,7 +927,7 @@ base class _NativeSemanticsUpdateBuilder extends NativeFieldWrapperClass1 implem
       hint,
       hintAttributes,
       tooltip,
-      textDirection != null ? textDirection.index + 1 : 0,
+      textDirection.index + 1,
```

Comment:
what happen to the case where textDirection used to be null? I think we want to send 0 to the native side?

Reply:
If the native side has to handle TextDirection being null, and it's a valid use case, then I think this PR doesn't make sense and can be closed. Wdyt?

Reply:
I only take a quick glance we only enforce TextDirection to be non-null when there is and text data in the SemanticsData. I think this pr may not be the right move

Reply:
I think a new test should be added, so when in the future someone tries to do the same PR as this one, that test would fail.

Reply:
This is weird I remember we used to have a framework smoke test, but it was gone from the ci

Reply:
It was disabled here https://github.com/flutter/engine/pull/50800